GVFS/GVFS.Common/InstallerPreRunChecker.cs

```diff
@@ -248,6 +258,12 @@ private bool IsGVFSUpgradeAllowed(out string error)
                 return false;
             }
 
+            if (!this.FreshInstallOrServiceRunning())
+            {
+                error = "GVFS Service is not running.\nStart \"GVFS.Service\" and run \"gvfs upgrade\" again.";

```

Comment:
>>Will this error show up on a fresh install
No. For fresh install using Upgrader tool directly, this would throw no errors.

>> Should we rephrase this method as if (!this.ServiceRunning()) and say no if it isn't even installed?
Rephrased it to IsServiceInstalledAndNotRunning.